### ISSUE TITLE
Declaring an array property breaks intellisense

### ISSUE THREAD

**Issue by rchl** [vuejs/vetur#1422]
- [x] I have searched through existing issues
- [x] I have read through [docs](https://vuejs.github.io/vetur)
- [x] I have read [FAQ](https://github.com/vuejs/vetur/blob/master/docs/FAQ.md)

## Info

- Platform: macOS
- Vetur version: 0.22.2
- VS Code version: 1.38.0

## Problem

Vetur's intellisense breaks when one of the props is defined as being of Array type.

Prop type is `String`:
![type-string](https://user-images.githubusercontent.com/153197/64555762-550f2600-d33e-11e9-91f9-90276efdbe00.png)

Prop type is `Array`:
![type-array](https://user-images.githubusercontent.com/153197/64555764-550f2600-d33e-11e9-9b7e-656b549534ed.png)

## Reproducible Case

1. Clone https://github.com/rchl/bug-vetur-prop-array
2. Run `npm install`
3. Open project in VS Code and open `components/test.vue` file

Expected that you don't see squiggly lines on many methods/properties. Only `notExisting` property in the template should show error as it doesn't exist.

## BTW

I know this bug is pretty much identical to the [old one that is now closed](1295) and was set as duplicate of #1242, but the conclusion there was that this is an [upstream bug](https://github.com/vuejs/vue/issues/9935) which upstream commented on and said that it's a typescript bug that is already fixed. But it appears that Vetur is forcing old typescript version (3.3) and that seems to be the bug here.

**Comment by shjyh** [comment]
same problem！
You can use the following code if use typescript：
``` typescript
props: {
    type: Array as ()=>Array<any>
}
```

**Comment by rchl** [comment]
For me it's JavaScript as seen in the screenshots. Maybe jsdoc would work but still...

**Comment by ktsn** [comment]
This is already fixed on TypeScript >= 3.5 and you can let Vetur use workspace TypeScript by setting `Vetur: Use Workspace Dependencies` config.

**Comment by rchl** [comment]
@ktsn yes, but why is plugin using old version by default? Shouldn't this issue stay open so that can be fixed?

**Comment by rchl** [comment]
There was ed9a8e5ed9a9a78b6bf72c5a2678afe16367a417 integrated quite recently to update TS, but I wonder if it upgraded wrong TS (meant to upgrade one in the server?).

**Comment by ktsn** [comment]
Because we don't have enough resource to update TypeScript version frequently & instantly. That's why we have 'Use workspace dependencies' config. In any cases, updating TypeScript version should be another issue as original problem can be solved by the config.

**Comment by rchl** [comment]
@ktsn I don't agree with you on that.

This issue happens in the default configuration. I've created it specifically for fixing the outdated typescript dependency issue.  

More so, this happens in default configuration with standard code. That, to me, qualifies as a bug and changing settings to fix it sounds more like a workaround.

And ed9a8e5ed9a9a78b6bf72c5a2678afe16367a417 was done to fix #1389 and it won't fix it (I'm almost certain) because wrong dependency was updated. So the intention was to update typescript anyway and just by accident it didn't happen yet.
